kafka-init/create_topics.py

```diff
@@ -161,13 +156,46 @@ def create_topics(default_config, existing_topics):
                 'partitions not set for topic %s, it will not be created!',
                 topic_name)
             continue
-
+        if topic_name in existing_topics:
+            logger.info('Topic %s already exists, ensuring configuration options match up',
+                        topic_name)
+            existing_topics_config[topic_name] = configs
+            continue
         logger.info('Creating topic %s: partitions=%s, replicas=%s, config=%r',
                     topic_name, partitions, replicas, configs)
         create_topic(topic_name, partitions, replicas, configs)
         created.append(topic_name)
 
-    return created
+    return created, existing_topics_config
+
+
+def update_topic_configs(existing_topics_config):
+    for topic, config in existing_topics_config.iteritems():
+        topic_describe_output, err = kafka_topics("describe", ["--topic", topic])
+        topic_describe = topic_describe_output.split('\n')[0]
+        current_config = {}
+        if "Configs" in topic_describe:
+            topic_config = topic_describe.split("Configs:")[1]
+            if topic_config:
+                for config_str in topic_config.split(","):
+                    k, v = map(lambda s: s.strip(), config_str.split('=', 2))
+                    current_config[k] = v
+
+        configs_to_delete = filter(lambda x: x not in config.keys(), current_config.keys())
+        if configs_to_delete:
+            logger.info('Removing topic configuration options %s from topic %s', configs_to_delete, topic)
+            config_input = []
+            for config_opt in configs_to_delete:
+                config_input.append("--delete-config")
+                config_input.append(config_opt)
+            kafka_topics("alter", ["--topic", topic] + config_input)
+
+        if config:
+            logger.info('Adding/Updating topic configuration options %s to topic %s', config.keys(), topic)
```

Comment:
It always calls alter on the topic even if the config options already match. Probably we don't care about that.

Reply:
yeah should hvae no harm